diff -dur -x '*~' xfce4-places-plugin-1.1.0.orig/configure.in xfce4-places-plugin-1.1.0/configure.in
--- xfce4-places-plugin-1.1.0.orig/configure.in	2008-06-07 20:01:20.000000000 +0200
+++ xfce4-places-plugin-1.1.0/configure.in	2008-10-29 10:45:30.000000000 +0100
@@ -28,7 +28,7 @@
 XDT_CHECK_PACKAGE([EXO], [exo-0.3], [0.3.1.1])
 
 dnl check for i18n support
-XDT_I18N([ar be ca cs da de dz el en_GB eo es et eu fi fr gl he hu it ja ka ko lt lv mk nb_NO nl pa pl pt_BR pt_PT ro ru sq sv tr uk zh_CN zh_TW])
+XDT_I18N([ar be ca cs da de dz el en_GB eo es et eu fi fr gl he hu it ja ka ko lt lv mk nb nl pa pl pt_BR pt ro ru sq sv tr uk zh_CN zh_TW])
 
 XDT_FEATURE_DEBUG()
 
diff -dur -x '*~' xfce4-places-plugin-1.1.0.orig/po/LINGUAS xfce4-places-plugin-1.1.0/po/LINGUAS
--- xfce4-places-plugin-1.1.0.orig/po/LINGUAS	2008-06-07 20:01:09.000000000 +0200
+++ xfce4-places-plugin-1.1.0/po/LINGUAS	2008-10-29 10:45:16.000000000 +0100
@@ -1,2 +1,2 @@
 # set of available languages (in alphabetic order)
-ar be ca cs da de dz el en_GB eo es et eu fi fr gl he hu it ja ka ko lt lv mk nb_NO nl pa pl pt_BR pt_PT ro ru sq sv tr uk zh_CN zh_TW
+ar be ca cs da de dz el en_GB eo es et eu fi fr gl he hu it ja ka ko lt lv mk nb nl pa pl pt_BR pt ro ru sq sv tr uk zh_CN zh_TW
